src/DotNet.Testcontainers/Containers/ResourceReaper.cs

```diff
@@ -135,27 +134,47 @@ public static Task<ResourceReaper> GetAndStartNewAsync(string ryukImage = RyukIm
     [PublicAPI]
     public static async Task<ResourceReaper> GetAndStartNewAsync(Guid sessionId, string ryukImage = RyukImage, TimeSpan initTimeout = default, CancellationToken ct = default)
     {
+      var ryukInitializedTaskSource = new TaskCompletionSource<bool>();
+
       var resourceReaper = new ResourceReaper(sessionId, ryukImage);
 
+      if (!TestcontainersSettings.ResourceReaperEnabled)
+      {
+        return resourceReaper;
```

Comment:
Returning a `ResourceReaper` instance when `ResourceReaper` is not enabled is a little weird. This would be the only case in which `GetAndStartNewAsync` returns a `ResourceReaper` that is actually not initialized.

Maybe `TestcontainersSettings.ResourceReaperEnabled` should only disable the default `ResourceReaper` and `GetAndStartDefaultAsync` should return null if it was not enabled.

Reply:
I agree. I'm not happy with it, too. The only reason I added it was: `Initialization.StopAsync`. This will trigger the default Resource Reaper clean up immediately. For Windows, it will fail, because we call `ResourceReaper.GetAndStartDefaultAsync` explicit for the first time and it can't start Ryuk.